### PR TITLE
feat: Improve hono middleware type

### PR DESCRIPTION
**Pull request by G3root**
## Type of change

- [x] 📖 Refactoring / dependency upgrade / documentation

## Description

this pr improves honojs middleware types

### CODE REVIEW

File: packages/hono/src/index.ts [hunk 1]
```diff
@@ -42,8 +49,8 @@ export type UnkeyConfig<TContext = Context> = {
   onError?: (c: TContext, err: UnkeyError) => Response | Promise<Response>;
 };
 
-export function unkey(config?: UnkeyConfig): MiddlewareHandler {
-  return async (c, next) => {
+export function unkey(config?: UnkeyConfig) {
+  return createMiddleware(async (c, next) => {
```
Comment (G3root) [inline packages/hono/src/index.ts:53]: use middleware helper to seamlessly interact with various platforms 

https://hono.dev/helpers/adapter#adapter-helper

File: packages/hono/src/index.ts [hunk 2]
```diff
@@ -69,5 +76,9 @@ export function unkey(config?: UnkeyConfig): MiddlewareHandler {
 
     c.set("unkey", res.result);
     await next();
-  };
+  });
 }
+
+export const getUnkey = (c: Context) => {
```
Comment (G3root) [inline packages/hono/src/index.ts:82]: helper to get unkey context

**Comment by chronark** [comment]
@G3root could you explain why this is superior to `c.get("unkey")`

afaik `c.get("unkey")` is already fully typesafe

**Comment by G3root** [comment]
> @G3root could you explain why this is superior to `c.get("unkey")`
> 
> afaik `c.get("unkey")` is already fully typesafe

Agree it's already typesafe and this is simply a matter of preference. This pattern is utilized by some third-party Hono middlewares, such as Clerk, Sentry, etc... so I added it

**Comment by chronark** [comment]
Let's go with simpler and less magic here, `c.get("unkey")` is fine and what they use in their official docs.

**Pull request closed without merging**